crates/criterion_compat/criterion_fork/src/routine.rs

```diff
@@ -191,7 +191,9 @@ pub(crate) trait Routine<M: Measurement, T: ?Sized> {
             .unwrap();
         }
 
+        let _guard = codspeed::fifo::BenchGuard::new_with_runner_fifo();
         let m_elapsed = self.bench(measurement, &m_iters, parameter);
+        core::mem::drop(_guard);
```

Comment:
Mostly cosmetic, but to me, the whole point of creating a guard is not to have to call the drop manually. Otherwise, bare enter()/exit() function would do the job more explicitly (instead of core::mem::drop). 

So the biggest benefit I see with guard is to use it like this:
```suggestion
        let m_elapsed = {
            let _guard = codspeed::fifo::BenchGuard::new_with_runner_fifo();
            self.bench(measurement, &m_iters, parameter)
        }
```

Is there any reason I'm missing for not doing this instead?

Reply:
The main reason is making it more explicit and show _where_ we drop it. I did it like this in divan because there we have a huge while loop that we want to profile: 
https://github.com/CodSpeedHQ/codspeed-rust/blob/788b13d78ccbac5e1860bf271cba8c34693ec3f9/crates/divan_compat/divan_fork/src/bench/mod.rs#L660-L814

But I agree, adding a scope for this simple scenario makes more sense :+1:

Reply:
Ah, yes, definitely agree for larger blocks of code. If you can refactor for the smaller one, it would be great!